src/Microsoft.AspNetCore.Razor/CodeGenerators/CSharpTagHelperCodeRenderer.cs

```diff
@@ -287,6 +287,12 @@ protected virtual string GenerateUniqueId()
                         continue;
                     }
 
+                    if (attributeValueChunk is PreallocatedTagHelperAttributeChunk)
+                    {
+                        RenderPreallocatedBoundAttribute(attributeName, attributeValueChunk);
+                        continue;
```

Comment:
You are correct. Initially we thought it would be sufficient to just add the preallocated alltributes to the AllAttributes list and it is not need to set the associated tag helper properties. But after debugging, I found that we need to set the tag helper properties to the preallocated attribute values. I will make the change.